### PR TITLE
Content skeleton

### LINKED ISSUE #210: [RTC] Content Skeleton
User Story:
As a site owner, I want a content skeleton in place that follows best practices, so I can begin using AEM to author or migrate content into the site.

Details:
1. For multi-country/lang option, following should get auto-generated:

- The language master folder & top level page
- Live copy of the top level page in the specified language
- Blueprint

2. For a single country

- Top level page

Note - Use Templates as mentioned in #179 to generate the page content structure.

### PR DESCRIPTION
**Pull request by bpauli**
Add new parameters `isSingleCountryWebsite` and `language_country` to the archetype metadata. Depending on the user input the example content structure will provide a language-master with blueprint configuration or only a single country content structure.

fixes #210

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-| Copyright 2015 Adobe Systems Incorporated
+| Copyright 2015 Adobe
```
Comment (jckautzmann) [inline pom.xml]: The naming change only applies to new files.

File: src/main/resources/META-INF/archetype-post-generate.groovy [hunk 2]
```diff
@@ -30,4 +25,47 @@ if (optionAemVersion == "6.3.3") {
...
+
+    if (isSingleCountryWebsite == "y") {
+        def languageMastersDir = new File(uiContentPackage, "src/main/content/jcr_root/content/${contentFolderName}/language-masters")
+        languageMastersDir.deleteDir()
+        def msmAppsDir = new File(uiAppsPackage, "src/main/content/jcr_root/apps/msm")
+        msmAppsDir.deleteDir()
+    } else {
+        def languageDir = new File(uiContentPackage, "src/main/content/jcr_root/content/${contentFolderName}/language-masters/en")
+        languageDir.renameTo(new File(uiContentPackage, "src/main/content/jcr_root/content/${contentFolderName}/language-masters/${contentDetails[0]}"))
+
```
Comment (jckautzmann) [inline src/main/resources/META-INF/archetype-post-generate.groovy]: unneeded blank line

File: src/main/resources/META-INF/archetype-post-generate.groovy [hunk 3]
```diff
@@ -30,4 +25,47 @@ if (optionAemVersion == "6.3.3") {
 if (optionIncludeFrontendModule == "n") {
     assert new File(rootDir, "ui.frontend").deleteDir()
     removeModule(rootPom, "ui.frontend")
+}
+
+buildContentSkeleton()
+
+/**
+ * Create content skeleton based upon isSingleCountry
```
Comment (jckautzmann) [inline src/main/resources/META-INF/archetype-post-generate.groovy]: Create -> Creates
Put the text in one line

File: README.md [hunk 4]
```diff
@@ -91,6 +91,7 @@ optionAemVersion            |  6.5.0  | Target AEM version
 optionIncludeExamples       |    y    | Include a Component Library example site
 optionIncludeErrorHandler   |    n    | Include a custom 404 response page
 optionIncludeFrontendModule |    n    | Include a dedicated frontend module
+isSingleCountryWebsite      |    y    | Create language-master structure in example content
```
Comment (bstopp) [inline README.md]: The language_country flag is missing.

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/.content.xml [hunk 5]
```diff
@@ -9,11 +9,13 @@
         sling:redirect="{Boolean}true"
         sling:redirectStatus="{Long}302"
         sling:resourceType="foundation/components/redirect"
-        redirectTarget="/content/${contentFolderName}/en">
+        redirectTarget="/content/${contentFolderName}/${language_country.substring(3)}/${language_country.substring(0,2)}">
```
Comment (bstopp) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/.content.xml]: Don't assume country/language are only 2 characters:

```suggestion
        redirectTarget="/content/${contentFolderName}/${language_country.split('_').get(1)}/${language_country.split('_').get(0)}">
```

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/language-masters/.content.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Page">
+    <jcr:content
+        cq:lastModified="{Date}2015-10-12T23:10:09.098+02:00"
+        cq:lastModifiedBy="admin"
+        cq:redirectTarget="/content/${contentFolderName}/${language_country.substring(3)}/${language_country.substring(0,2)}"
```
Comment (bstopp) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/language-masters/.content.xml]:
```suggestion
        cq:redirectTarget="/content/${contentFolderName}/${language_country.split('_').get(1)}/${language_country.split('_').get(0)}"
```

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/language-masters/.content.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+        cq:template="/conf/${confFolderName}/settings/wcm/templates/homepage"
+        jcr:primaryType="cq:PageContent"
+        jcr:title="Language Masters"
+        sling:resourceType="foundation/components/redirect"
+        hideInNav="{Boolean}true"/>
+    <${language_country.substring(0,2)}/>
```
Comment (bstopp) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/language-masters/.content.xml]:
```suggestion
    <${language_country.split('_').get(0)}/>
```

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/.content.xml [hunk 8]
```diff
@@ -9,11 +9,13 @@
...
         <image jcr:primaryType="nt:unstructured">
             <file/>
         </image>
     </jcr:content>
-    <en/>
-    <fr/>
+#if ( $isSingleCountryWebsite == "n" )
+    <language-masters/>
+#end
+    <${language_country.substring(3)}/>
```
Comment (bstopp) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/.content.xml]:
```suggestion
    <${language_country.split('_').get(1)}/>
```

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/us/.content.xml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Page">
+    <${language_country.substring(0,2)}/>
+    <jcr:content
+        cq:lastModified="{Date}2016-03-07T17:32:10.926+01:00"
+        cq:lastModifiedBy="admin"
+        cq:redirectTarget="/content/${contentFolderName}/${language_country.substring(3)}/${language_country.substring(0,2)}"
```
Comment (bstopp) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/us/.content.xml]:
```suggestion
        cq:redirectTarget="/content/${contentFolderName}/${language_country.split('_').get(1)}/${language_country.split('_').get(0)}"
```

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/us/en/.content.xml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+            sling:resourceType="wcm/foundation/components/responsivegrid">
+            <responsivegrid
+                jcr:lastModifiedBy="admin"
+                jcr:primaryType="nt:unstructured"
+                sling:resourceType="wcm/foundation/components/responsivegrid"/>
+        </root>
+#if ( $isSingleCountryWebsite == "n" )
+        <cq:LiveSyncConfig
+            cq:isDeep="{Boolean}true"
+            cq:master="/content/${contentFolderName}/language-masters/${language_country.substring(0,2)}"
```
Comment (bstopp) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/__contentFolderName__/us/en/.content.xml]:
```suggestion
            cq:master="/content/${contentFolderName}/language-masters/${language_country.split('_').get(0)}"
```

File: src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 11]
```diff
@@ -72,6 +72,21 @@
         <requiredProperty key="siteName">
             <defaultValue>${appsFolderName}</defaultValue>
         </requiredProperty>
+        
+        <!--
+            ISO 639-1 language / country code to create initial content structure
+         -->
+        <requiredProperty key="language_country">
+            <validationRegex>[a-z]+_[a-z]+</validationRegex>
```
Comment (bstopp) [inline src/main/resources/META-INF/maven/archetype-metadata.xml]: This should be more specific to supported ISO structure.
```suggestion
            <validationRegex>[a-z]{2,3}_[a-z]{2,3}</validationRegex>
```

**Pull request merged**